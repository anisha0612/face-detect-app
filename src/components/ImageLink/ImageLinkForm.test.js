import React from "react";
import { shallow } from "enzyme";
import { create } from "react-test-renderer";
import ImageLinkForm from "./ImageLinkForm";

describe("ImageLink Form Component", () => {
  // check  onchange the onInputChange function (prop) is called
  it("Input calls the onInputChange prop", () => {
    const mockOnInputChange = jest.fn();
    const event = {
      preventDefault() {},
      target: {
        value: "image.jpg",
      },
    };
    const wrapper = shallow(
      <ImageLinkForm onInputChange={mockOnInputChange} />
    );

    wrapper.find("input").simulate("change", event.target.value);
    expect(mockOnInputChange).toBeCalledWith("image.jpg");
  });

  //   Snapshot Testing of Image Link Component
  it("expect Image Link component is rendered", () => {
    const mockOnInputChange = jest.fn();
    const mockButtonSubmit = jest.fn();
    const wrapper = create(
      <ImageLinkForm
        onInputChange={mockOnInputChange}
        onButtonSubmit={mockButtonSubmit}
      />
    ).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
