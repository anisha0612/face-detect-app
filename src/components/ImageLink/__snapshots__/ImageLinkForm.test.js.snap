// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageLink Form Component expect Image Link component is rendered 1`] = `
<div>
  <p
    className="f3 center pa4"
  >
    This app recognizes face from images. Check it out!
  </p>
  <div
    className="center"
  >
    <div
      className="form center pa3 shadow-5 br3"
    >
      <input
        className="f4 pa2 w-70 center"
        onChange={[MockFunction]}
        type="text"
      />
      <button
        className="w-30 grow f4 link dib bg-purple white pv2 ph3 "
        onClick={[MockFunction]}
      >
        Detect
      </button>
    </div>
  </div>
</div>
`;
